servers/services/src/main/java/org/projectnessie/services/impl/ContentApiImpl.java

```diff
@@ -86,12 +93,25 @@ public GetMultipleContentsResponse getMultipleContents(
               .map(e -> ContentWithKey.of(toContentKey(e.getKey()), e.getValue()))
               .collect(Collectors.toList());
 
-      return ImmutableGetMultipleContentsResponse.builder().contents(output).build();
+      return GetMultipleContentsResponse.of(output, makeReference(ref));
     } catch (ReferenceNotFoundException ex) {
       throw new NessieReferenceNotFoundException(ex.getMessage(), ex);
     }
   }
 
+  private static Reference makeReference(WithHash<NamedRef> refWithHash) {
```

Comment:
A very similar method exists in `TreeApiImpl`... Shall we move the code into the shared `RefUtil`?

Reply:
they look similar - but are not really